### ISSUE TITLE
GetOrAdd(...) fails after GetOrAddAsync(...)

### ISSUE THREAD

**Issue by baddonkey** [alastairtree/LazyCache#12]
Please,  consider the following failing sample unit test:
```
        [Test]
        public async Task GetOrAddFollowinGetOrAddAsyncTryOut()
        {
            Func<Task<ComplexTestObject>> fetchAsync = () => Task.FromResult(testObject);
            await sut.GetOrAddAsync(TestKey, fetchAsync);
            var actualAsync = await sut.GetAsync<ComplexTestObject>(TestKey);
            Assert.IsNotNull(actualAsync);
            Assert.That(actualAsync, Is.EqualTo(testObject));

            Func<ComplexTestObject> fetchSync = () => testObject;
            sut.GetOrAdd(TestKey, fetchSync);
            var actualSync = sut.Get<ComplexTestObject>(TestKey);
            Assert.IsNotNull(actualSync); // actualSync == null
        }
```

Note: 

- Fails on last assertion as ```actualSync``` is ```null```.
- Having the ```GetOrAdd``` before ```GetOrAddAsync```, the test would succeed.

I assume that the private method ```UnwrapLazy``` is not considering the new ```AsyncLazy``` enhancements from September. Would be great if you could apply a fix for that, I would like to use this library in a mixed sync/async environment.

Thanks for sharing this library, great work.

Best regards, BadDonkey

**Comment by alastairtree** [comment]
Fixed in version 0.7.1, see https://www.nuget.org/packages/lazycache

**Comment by baddonkey** [comment]
Great, thanks!
